core/src/raw/http_util/bytes_range.rs

```diff
@@ -188,13 +181,14 @@ where
             Bound::Excluded(n) => n + 1,
             Bound::Unbounded => 0,
         };
+
         let size = match range.end_bound().cloned() {
-            Bound::Included(n) => Some(n + 1 - offset),
-            Bound::Excluded(n) => Some(n - offset),
-            Bound::Unbounded => None,
+            Bound::Included(n) => n - offset + 1,
+            Bound::Excluded(n) => n - offset,
+            Bound::Unbounded => panic!("end of range must be specified"),
```

Comment:
@Xuanwo I'm thinking if we should implement `TryFrom` instead here.

Reply:
I found that we have a place rely on this:

```rust
/// Set the range header for this operation.
pub fn range(self, range: impl RangeBounds<u64>) -> Self {
    self.map(|(args, op_reader)| (args.with_range(range.into()), op_reader))
}
```

I feel like We should remove the `impl From` for BytesRange, and change the `FutureRead::range`'s implemantion.